tests/rbd/rbd_clone_trash_restore_parent_image.py

```diff
@@ -0,0 +1,173 @@
+        size_before_io = rbd_du["images"][0]["used_size"]
+        log.info(f"Clone size before IO : {size_before_io}")
+
+        client = kw["ceph_cluster"].get_nodes(role="client")[0]
+        log.info(
+            "Move parent image to trash when IO on Clone is in progress. (use rbd bench)"
+        )
+        with parallel() as p:
+            p.spawn(
+                run_fio,
+                image_name=clone,
+                pool_name=pool,
+                client_node=client,
+                size="1G",
+            )
+            p.spawn(
+                move_to_trash_and_restore,
+                rbd=rbd,
+                pool=pool,
+                image=image,
+                sleep_time=60,
+            )
+
+        rbd_du = rbd.get_disk_usage_for_pool(pool, image=clone)
+        size_after_io = rbd_du["images"][0]["used_size"]
+        log.info(f"Clone size after IO : {size_after_io}")
+
+        written = int(
+            (size_after_io - size_before_io) / 1073741824
+        )  # 1GB = 1073741824 bytes
+        if written != 1:
+            log.error(
+                f"1GB of IOs were intended to be written on clone but only {written}GB"
+                "were written, so there might have been some interruption in IO"
+            )
+            return 1
+
+        rbd_info = rbd.image_info(pool, clone)
+        if (
+            rbd_info.get("parent").get("image") == image
+            and rbd_info.get("parent").get("snapshot") == snap
+            and not rbd_info.get("parent").get("trash")
+        ):
+            log.info(
+                f"Parent-clone {image}-{clone} relationship is intact and parent is restored from trash"
+            )
+        else:
+            log.error(f"Parent-clone {image}-{clone} relationship broken")
+            return 1
+
+        return 0
+    except Exception as error:
+        log.error(str(error))
+        return 1
+
+    finally:
+        rbd.clean_up(pools=[kw["config"][pool_type]["pool"]])
+
+
+def run(**kw):
+    """Verify parent image moved to trash and restored back when IO running on clone
+    and parent clone relationship still active.
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common package,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-11414 - In a parent clone relationship, move parent image to trash.
+    Dont sever the child image from parent, but try to restore the parent, and
+    verify the clone relationship is intact. (Keep IO in progress on clone image)
+    Test Case Flow
+    1. Create pool, image and clone the image. Start IOs on clone.
+    2. Move the parent image to trash.
+    3. Restore the parent  image back
+    4. Check parent-clone relationship.(Do rbd info on clone)
+    """
+    rbd_obj = initial_rbd_config(**kw)
+    if rbd_obj:
```

Comment:
Before starting test case execution add `log.info` for the test case description along with polarion id.

Reply:
Done.